### ISSUE TITLE
More information in jobs resource response

### ISSUE THREAD

**Issue by sfdc-kho** [vmware-archive/salt-api#49]
Currently, the minion response when a job hasn't completed is:
bash-3.2$ curl -i localhost:8000/jobs/20130225115005304834 \

> ```
>     -H "X-Auth-Token: 129c6a5dadd1463932fd9db452b52148c44c6c91" \
> ```
> 
>    -H "Accept: application/json"
> HTTP/1.1 200 OK
> Content-Length: 16
> Vary: Accept-Encoding
> Server: CherryPy/3.2.2
> Allow: GET, HEAD, POST
> Cache-Control: private
> Date: Mon, 25 Feb 2013 19:50:12 GMT
> Content-Type: application/json
> Set-Cookie: session_id=129c6a5dadd1463932fd9db452b52148c44c6c91; expires=Tue, 26 Feb 2013 05:50:12 GMT; Path=/

{"return": [{}]}

I am using this and a combination of saltutil.running to determine whether the job is still running, dead or completed. However, if the other issues like streaming responses get incorporated, it will become increasingly difficult to detect the state of any given job. Is it possible to return more information about what the state of the job is in the response?

**Comment by whiteinge** [comment]
This is tricky since Salt itself cannot determine when a job is "complete". That said you can safely rely on the `/jobs/<id>` interfaces to not change. If we do implement something like streaming responses we will do so at another URL point.

**Comment by whiteinge** [comment]
To rephrase the above a bit: There's no additional information on an executing job on the Salt master (the `/jobs/<id>` URLs just wrap Salt's `job` runner module). The method you are already using to fetch a list of what minions are running what jobs is the best way to see what jobs are still being executed and where.

The `/jobs/<id>` URL is intended to be a convenience URL to provide a nicer interface than calling the individual commands directly. It could be quite useful to set up another convenience URL that combines both commands somehow and compares job IDs.

**Comment by sfdc-kho** [comment]
Agreed, I wasn't suggesting that salt-api should somehow create additional data that salt is not inherently providing. I do like your second suggestion that there be another convenience resource so that callers don't have to manually combine different data points to get a complete picture.

**Comment by cro** [comment]
BugzID:49
